Automattic/wp-calypso
# Update/shared-setup-form-component

### Notes

This PR is related to #https://github.com/Automattic/wp-calypso/issues/67974. I would like to extract a common SetupForm component before adding the Newsletter Post Setup flow/screen, which will use the same form code.

The already-merged [PR for the Link in Bio Post-Setup screen](https://github.com/Automattic/wp-calypso/pull/67835) also has some relevant context. In that PR, the Link in Bio Setup and Post-Setup screens are already using/importing same form code. But that PR left the form code in the link-in-bio-setup directory rather than putting in a broader, shared directory. This PR moves that common form to a shared directory, updates it to handle accent color, and updates the newsletter setup screen to use the same form as well. That PR also has some discussion on why we have different Setup vs Post-Setup screens for each flow in the first place. 

### Proposed Changes

We currently have three tailored flow screens that effectively use the same Setup form: link-in-bio-setup, link-in-bio-post-setup, newsletter-setup. I will be adding a fourth shortly, newsletter-post-setup (see https://github.com/Automattic/wp-calypso/issues/67974). And we will likely be adding more for other tailored flows. To reduce code duplication, this PR creates a common SetupForm component that all these screens can use. 

Specific changes: 
* Create a common SetupForm component. This includes a separate child component for AccentColorControl, which will display conditionally for flows like newsletter that need it. This also includes a style.scss file, and any styles related to the setup form have been moved from other stylesheets to this new stylesheet.
* Update Link in Bio Setup screen to use this new form. 
* Update Link in Bio Post Setup screen to use this new form. 
* Update Newsletter Setup screen to use this new form. 

### Testing Instructions

This PR simply refactors how components / screens are composed. It should have no effect on logic, functionality, or user interface. To test, we will want to test the Link in Bio and Newsletter flows and confirm there are now regressions/breakages.

For reference, here is a comparison of the before/after forms for Newsletter and Link in Bio: 
![yzVwOG.gif](https://user-images.githubusercontent.com/21228350/193106860-8c9b8bed-5e55-4437-8786-eb4609d1ef89.gif)

1) Review code changes and confirm no concerns with concept and code organization.

2) Checkout this branch and run yarn and yarn start if needed. 

3) Test Newsletter flow: 
   - Start at https://wordpress.com/hp-2022-tailored-flows/ and select Newsletter flow. 
   - Before or on the first Setup screen (with title/tagline/icon/color), be sure to replace http://wordpress.com/ with http://calypso.localhost:3000/ so you are running local branch. 
   - TEST: Confirm that Newsletter setup screen and form load correctly, including styling and any web copy or placeholder text. Fill out form and continue. 
   - TEST: When you arrive at Launchpad, confirm that your title, tagline, icon, and accent color are all correctly rendered in the web preview. 

4) Test Link in Bio flow: 
   - Start at https://wordpress.com/hp-2022-tailored-flows/ and select Link in Bio flow. 
   - Before or on the first Setup screen (with title/tagline/icon), be sure to replace http://wordpress.com/ with http://calypso.localhost:3000/ so you are running local branch. 
   - TEST: Confirm that Link in Bio setup screen and form load correctly, including styling and any web copy or placeholder text Fill out form and continue. 
   - TEST: When you arrive at Launchpad, confirm that your title, tagline, and icon are all correctly rendered in the web preview. 
   - Click 'Personalize Link in Bio' to go to the post-setup screen. 
   - TEST: Again confirm that setup screen and form load correctly, including styling and any web copy or placeholder text.
   - Make some changes and click continue to return to Launchpad. 
   - TEST: When you return to Launchpad, confirm that any changes you made are accurately rendered in the web preview.
